src/main/java/com/openshift/jenkins/plugins/pipeline/dsl/OpenShiftBuilder.java

```diff
@@ -153,9 +166,38 @@ public Step newInstance(Map<String, Object> arguments) throws Exception {
                     step.setWaitTime(waitTime.toString());
                 }
             }
+            if (arguments.containsKey(("waitUnit"))) {
+                Object waitUnit = arguments.get("waitUnit");
+                if (waitUnit != null) {
+                    step.setWaitUnits(waitUnit.toString());
+                }
+            }
+
             ParamVerify.updateDSLBaseStep(arguments, step);
             return step;
         }
+
+        public ListBoxModel doFillWaitUnitItems() {
+            ListBoxModel items = new ListBoxModel();
+            switch(GlobalConfig.getBuildWaitUnits()){
+                case "minutes":
+                    items.add("Minutes", "minutes");
+                    items.add("Milliseconds", "milliseconds");
+                    items.add("Seconds", "seconds");
+                    break;
+                case "seconds":
+                    items.add("Seconds", "seconds");
+                    items.add("Milliseconds", "milliseconds");
+                    items.add("Minutes", "minutes");
+                    break;
+                case "milliseconds":
+                    items.add("Milliseconds", "milliseconds");
+                    items.add("Seconds", "seconds");
+                    items.add("Minutes", "minutes");
+                    break;
+            }
+            return items;
+        }
```

Comment:
Do you mean between the dsl/OpenShiftBuilder.java and pipeline/OpenShiftBuilder.java? I assumed since there were two Descriptors we needed the method in both.  I can remove it if I'm wrong